### PR TITLE
SWARM-1115 - @DefaultDeployment should ensure persistence.xml is loca…

### PR DESCRIPTION
**Pull request by bobmcwhirter**
…ted right.

Motivation
----------

For .wars, src/main/resources/META-INF/persistence.xml is weird and should
end up in WEB-INF/classes/META-INF/persistence.xml.

Modifications
-------------

Change DefaultDeployment to handle the special-case of persistence.xml.

Result
------

@DefaultDeployment is slightly more useful.

- [X] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [X] Have you created a [JIRA](https://issues.jboss.org/browse/SWARM) and used it in the commit message?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/wildfly-swarm/wildfly-swarm/pulls) for the same issue?
- [X] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

**Comment by emag** [comment]
This commit covers only persistence.xml? Sorry for not referring to it and the bad title in SWARM-1115, some JPA examples point a sql file for auto-loading(like import.sql).
I think if the app packaging is war, the all assets in src/main/resources should be put under WEB-INF/classes.

**Comment by bobmcwhirter** [comment]
okay, I can do that. I wasn't sure.

File: arquillian/adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/DefaultDeploymentScenarioGenerator.java [hunk 1]
```diff
@@ -154,6 +154,16 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
         return Collections.singletonList(description);
     }
 
+    protected String handleExceptionalCases(Archive archive, String input) {
+        if (archive.getName().endsWith(".war")) {
+            if (input.startsWith("META-INF/")) {
```
Comment (emag) [inline arquillian/adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/DefaultDeploymentScenarioGenerator.java]: I think not only META-INF dir, but also all 'src/main/resources' things should be put in WEB-INF/classes.

Comment (emag) [reply arquillian/adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/DefaultDeploymentScenarioGenerator.java]: Ah, but maybe need to exclude project-\<name\>.yml ...

Comment (bobmcwhirter) [reply arquillian/adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/DefaultDeploymentScenarioGenerator.java]: Or maybe we put it all under WEB-INF/classes, and make our yaml-finding bits smart enough to look there.

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**